errors.go

```diff
@@ -215,6 +235,41 @@ func (errorType ErrorType) UserErrorf(msg string, args ...interface{}) error {
 	}
 }
 
+// DetailsAdd adds a details element to an error
+func (errorType ErrorType) DetailsAdd(err error, details interface{}) error {
+	if err == nil {
+		return errorType.Details(details)
+	}
+
+	c := new(customError)
+	c.error = errors.WithStack(err)
+	c.userMessage = GetUserMessage(err)
+
+	origDetails := GetDetails(err)
+	if origDetails != nil {
+		c.details = append(origDetails, details)
+	} else {
+		c.details = []interface{}{details}
+	}
```

Comment:
ok. back to my original comment regarding `GetDetails`
If it would `return []interface{}{}` I think we could remove this `if` block and always append.
Am I right? or maybe this requires a `make([]interface{})` and in that case, it's not worth the change.

@cben @tzvatot

Reply:
This doesn't require a `make`, in fact, just returning `[]interface{}{}` in `GetDetails` removes this entire chunk into a one-liner